File: src/test/java/main/TestServer.java
package main;

import java.util.Arrays;
import java.util.List;

import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import net.minestom.server.MinecraftServer;
import net.minestom.server.entity.Player;
import net.minestom.server.event.GlobalEventHandler;
import net.minestom.server.event.player.PlayerLoginEvent;
import net.minestom.server.event.player.PlayerSpawnEvent;
import net.minestom.server.instance.Chunk;
import net.minestom.server.instance.ChunkGenerator;
import net.minestom.server.instance.ChunkPopulator;
import net.minestom.server.instance.InstanceContainer;
import net.minestom.server.instance.batch.ChunkBatch;
import net.minestom.server.instance.block.Block;
import net.minestom.server.inventory.PlayerInventory;
import net.minestom.server.item.ItemStack;
import net.minestom.server.item.Material;
import net.minestom.server.utils.Position;
import net.minestom.server.world.biomes.Biome;

public class TestServer {
	
	private static InstanceContainer INSTANCE;
	private static final Position SPAWN_POSITION = new Position(0, 100, 0);
	
    public static void main(String[] args) {
    	MinecraftServer server = MinecraftServer.init();
    	
    	INSTANCE = MinecraftServer.getInstanceManager().createInstanceContainer();

    	INSTANCE.setChunkGenerator(new TestGenerator());
    	
    	GlobalEventHandler handler = MinecraftServer.getGlobalEventHandler();
    	
    	handler.addEventCallback(PlayerLoginEvent.class, (event) -> event.setSpawningInstance(INSTANCE));
    	handler.addEventCallback(PlayerSpawnEvent.class, (event) -> {
    		
    		Player player = event.getPlayer();
    		
    		player.teleport(SPAWN_POSITION);
    		
    		PlayerInventory inventory = player.getInventory();
    		
    		inventory.addItemStack(ItemStack.of(Material.REDSTONE).withAmount(100));
    		inventory.addItemStack(ItemStack.of(Material.REDSTONE_BLOCK).withAmount(100));
    		inventory.addItemStack(ItemStack.of(Material.REPEATER).withAmount(100));
    		inventory.addItemStack(ItemStack.of(Material.BELL).withAmount(100));
    		inventory.addItemStack(ItemStack.of(Material.PISTON).withAmount(100));
    	});
    	
    	SkylaExtension.runStandalone();
    	
    	server.start("0.0.0.0", 25565);
    }
    
    private static class TestGenerator implements ChunkGenerator {
		@Override
		public void generateChunkData(@NotNull ChunkBatch batch, int chunkX, int chunkZ) {
			for (int x = 0; x < Chunk.CHUNK_SIZE_X; x++)
				for (int z = 0; z < Chunk.CHUNK_SIZE_Z; z++) {
					batch.setBlock(x, 80, z, Block.STONE);
				}
		}

		@Override
		public void fillBiomes(@NotNull Biome[] biomes, int chunkX, int chunkZ) {
			Arrays.fill(biomes, Biome.PLAINS);
		}

		@Override
		public @Nullable List<ChunkPopulator> getPopulators() {
			return null;
		}
    }
}

File: src/test/java/main/TestServer.kt
package main

import net.minestom.server.MinecraftServer
import net.minestom.server.event.player.PlayerLoginEvent
import net.minestom.server.event.player.PlayerSpawnEvent
import net.minestom.server.instance.Chunk
import net.minestom.server.instance.ChunkGenerator
import net.minestom.server.instance.ChunkPopulator
import net.minestom.server.instance.InstanceContainer
import net.minestom.server.instance.batch.ChunkBatch
import net.minestom.server.instance.block.Block
import net.minestom.server.item.ItemStack
import net.minestom.server.item.Material
import net.minestom.server.utils.Position
import net.minestom.server.world.biomes.Biome
import java.util.*

fun main(args: Array<String>) {
    val server = MinecraftServer.init()
    TestServer.INSTANCE = MinecraftServer.getInstanceManager().createInstanceContainer()
    TestServer.INSTANCE!!.chunkGenerator = TestServer.TestGenerator()
    val handler = MinecraftServer.getGlobalEventHandler()

    handler.addEventCallback(PlayerLoginEvent::class.java) { event: PlayerLoginEvent ->
        event.setSpawningInstance(
            TestServer.INSTANCE!!
        )
    }

    handler.addEventCallback(PlayerSpawnEvent::class.java) { event: PlayerSpawnEvent ->
        val player = event.player
        player.teleport(TestServer.SPAWN_POSITION)
        val inventory = player.inventory
        inventory.addItemStack(ItemStack.of(Material.REDSTONE, 100))
        inventory.addItemStack(ItemStack.of(Material.REDSTONE_BLOCK, 100))
        inventory.addItemStack(ItemStack.of(Material.REPEATER, 100))
        inventory.addItemStack(ItemStack.of(Material.BELL, 100))
        inventory.addItemStack(ItemStack.of(Material.PISTON, 100))
    }

    SkylaExtension.runStandalone()
    server.start("0.0.0.0", 25565)
}

object TestServer {
    internal var INSTANCE: InstanceContainer? = null
    internal val SPAWN_POSITION = Position(.0, 100.0, .0)

    internal class TestGenerator : ChunkGenerator {
        override fun generateChunkData(batch: ChunkBatch, chunkX: Int, chunkZ: Int) {
            for (x in 0 until Chunk.CHUNK_SIZE_X) for (z in 0 until Chunk.CHUNK_SIZE_Z) {
                batch.setBlock(x, 80, z, Block.STONE)
            }
        }

        override fun fillBiomes(biomes: Array<Biome>, chunkX: Int, chunkZ: Int) {
            Arrays.fill(biomes, Biome.PLAINS)
        }

        override fun getPopulators(): List<ChunkPopulator>? {
            return null
        }
    }
}